packages/kinvey-datastore2/src/operation.js

```diff
@@ -0,0 +1,14 @@
+export const OperationType = {
+  Create: 'Create'
+};
+Object.freeze(OperationType);
+
+export class Operation {
+  constructor(config = {}) {
```

Comment:
I don't like this way of constructing an Operation instance. If we are doing OOD, we should stick to OO. I don't think Java or C# would use this manner of constructing objects.

Reply:
My only concern was the arguments to the constructor would keep growing. I am fine changing it back.

Reply:
I know, 4+ constructor arguments feels odd. But I'd rather solve this differently. Perhaps the factory can instantiate an operation object and assign its properties the values it has been passed. Or we can branch off different Operation descendants. I've thought about it, but didn't feel pressing so I haven't done much in that direction.